app/rooms/preparation-room.ts

```diff
@@ -337,18 +339,23 @@ export default class PreparationRoom extends Room<PreparationState> {
 
   async onAuth(client: Client, options: any, request: any) {
     try {
-      super.onAuth(client, options, request)
       const token = await admin.auth().verifyIdToken(options.idToken)
       const user = await admin.auth().getUser(token.uid)
       const isBanned = await BannedUser.findOne({ uid: user.uid })
       const userProfile = await UserMetadata.findOne({ uid: user.uid })
       client.send(Transfer.USER_PROFILE, userProfile)
 
-      const isAlreadyInRoom = this.state.users.has(user.uid)
       const numberOfHumanPlayers = values(this.state.users).filter(
         (u) => !u.isBot
       ).length
-      if (numberOfHumanPlayers >= MAX_PLAYERS_PER_GAME && !isAlreadyInRoom) {
+
+      if (!this.state.uids.has(user.uid)) {
```

Comment:
this `uids` should not be needed, you can just check `state.users.some(u => u.uid === user.uid)`

Reply:
`this.state.users` is only updated in `OnJoinCommand`.

Reply:
doesn't change my point, to check if the user is already there, it should already have gone through the onjoin command

Reply:
No, onAuth comes before onJoin.

Reply:
Ok, I stand corrected after some testing.

Reply:
I know. But you're testing if the user has **already** joined here. Which means he is in state.users